django/middleware/csrf.py

```diff
@@ -105,19 +105,38 @@ def _reject(self, request, reason):
         )
         return _get_failure_view()(request, reason=reason)
 
+    def _get_current_token(self, request):
+        try:
+            if settings.CSRF_USE_SESSIONS:
+                return _sanitize_token(request.session[settings.CSRF_COOKIE_NAME])
```

Comment:
No, I don't think that is the way to go, as it would be very confusing and inconsistent to read in the settings file.